### ISSUE TITLE
tls/ssl support

### ISSUE THREAD

**Issue by ftao** [retr0h/ansible-etcd#5]
there is a config option
```
etcd_url_scheme    
``` 
If it is  set this to 'https',   we need a lot more options for proper tsl/ssl support . 
https://github.com/coreos/etcd/blob/master/Documentation/security.md
```
Client-to-server communication:

--cert-file=<path>: Certificate used for SSL/TLS connections to etcd. When this option is set, you can set advertise-client-urls using HTTPS schema.

--key-file=<path>: Key for the certificate. Must be unencrypted.

--client-cert-auth: When this is set etcd will check all incoming HTTPS requests for a client certificate signed by the trusted CA, requests that don't supply a valid client certificate will fail.

--trusted-ca-file=<path>: Trusted certificate authority.

Peer (server-to-server / cluster) communication:

The peer options work the same way as the client-to-server options:

--peer-cert-file=<path>: Certificate used for SSL/TLS connections between peers. This will be used both for listening on the peer address as well as sending requests to other peers.

--peer-key-file=<path>: Key for the certificate. Must be unencrypted.

--peer-client-cert-auth: When set, etcd will check all incoming peer requests from the cluster for valid client certificates signed by the supplied CA.

--peer-trusted-ca-file=<path>: Trusted certificate authority.
```

**Comment by retr0h** [comment]
@ftao I don't know much about this configuration.  if I were to set the url scheme to https, would I then need to always set the client ssl settings and peer ssl settings?  Just wanted to make sure I get the logic correct.

**Comment by ftao** [comment]
I thinks there are independent . 
You can only enable peer-to-peer encryption , without client-server encryption , and vice versa .
and It's reasonable in many cases  client will be inside trusted network (like localhost only or private network) ,  but communication between peers  are more likely cross un-trusted networks .

**Comment by retr0h** [comment]
@ftao see #6, although may not be exactly what we want.

**Comment by sdodson** [comment]
client and peer can be adjusted independently and I've found that peer side certs need to be signed for both server and client x509 v3 extended attributes where as the client side certs need only include server uses.

It'd probably be wise to make it easy to set at least the ca/crt/key independently, if not scheme all together. We've adapted your role in openshift-ansible and do just that, though this is all tailored for Red Hat packaging.

see
https://github.com/openshift/openshift-ansible/blob/master/roles/etcd/templates/etcd.conf.j2#L42-L52
https://github.com/openshift/openshift-ansible/blob/master/roles/etcd/defaults/main.yaml#L9-L14

Are you interested in including RHEL/Centos/Fedora support by way of installing existing etcd packages? (EL7 and current Fedoras have etcd 2.0.11-13). I'd be happy to contribute our changes back to your role if you're ok with including that.
